src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1381,7 +1381,22 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
     if ( digitizingFinished )
     {
       QgsGeometry g;
-      QgsCurve *curveToAdd = captureCurve()->clone();
+
+      //does compoundcurve contain circular strings?
+      //does provider support circular strings?
+      QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( layer() );
```

Comment:
You need to check whether layer is nullptr or not a vector layer -- both these situations are possible.

Reply:
Indeed, fixed